static/js/lib/api.js

```diff
@@ -6,6 +6,92 @@ import R from "ramda"
 
 import { S, parseJSON, filterE } from "./sanctuary"
 
+export const booksRequest = (force = false) => ({
+  url: "/api/library/books/",
+  update: {
+    books: (prev, next) => {
+      return next
+    },
+  },
+  transform: (responseJson) => {
+    // The response is a list of serialized books. This turns it into an object mapping
+    // the book id to the full serialized object: {[book.id]: [book]}
+    return {
+      books: R.fromPairs(
+        responseJson.map((book) => (
+          [book.id, book]
+        ))
+      )
+    }
+  },
+  force
+})
+
+export const deleteBookRequest = (bookId) => ({
+  url: `/api/library/books/${bookId}/`,
+  update:  {
+    books: (prev, next) => {
+      return R.omit(
+        [next.toDelete],
+        prev
+      )
+    },
+  },
+  transform: () => {
+    return {
+      books: {
+        toDelete: String(bookId)
+      }
+    }
+  },
+  options: {
+    method: "DELETE"
+  }
+})
+
+export const bookRequest = (bookId, force = false) => ({
+  url: `/api/library/books/${bookId}/`,
+  update: {
+    book: (prev, next) => {
+      return next
+    },
+  },
+  transform: (responseJson) => {
+    return { book: responseJson }
+  },
+  force
+})
+
+export const updateBookRequest = (bookId, payload) => ({
+  url: `/api/library/books/${bookId}/`,
+  body: payload,
+  update: {
+    book: (prev, next) => {
+      return next
+    },
+    books: (prev, next) => {
+      return R.merge(
+        prev,
+        next.toUpdate
+      )
+    }
+  },
+  transform: (responseJson) => {
+    const book = responseJson
+    return {
+      book: book,
+      books: {
+        toUpdate: {
+          [book.id]: book
+        }
+      }
+    }
+  },
+  options: {
+    method: "PATCH"
+  }
+})
```

Comment:
Some more "fancy" usage. Since this `PATCH` request returns a fully serialized version of the book we updated, we can simply update the book in the list of loaded books in the state.